Objects/floatobject.c

```diff
@@ -2233,21 +2233,23 @@ _PyFloat_Pack4(double x, unsigned char *p, int le)
 
     }
     else {
-        float y = (float)x;
-        const unsigned char *s = (unsigned char*)&y;
         int i, incr = 1;
 
-        if (Py_IS_INFINITY(y) && !Py_IS_INFINITY(x))
```

Comment:
I don't understand why you remove Py_IS_INFINITY(y).

Reply:
`y` is no longer defined at this point.